doc/release-process.md

```diff
@@ -63,21 +65,28 @@ progress bar displayed during the build process.
 
 ## Release process
```

Comment:
Before this, we need to document the new release stabilisation process:

- Release manager creates a branch from the target commit named `version-X.Y.0` (note: without leading `v`, and no `-foo` suffix), and pushes this to `zcash/zcash`.
- Release manager immediately runs the release script on this branch, to create `release-vX.Y.0-rc1`.
  - The script could go back to verifying the branch name if we wanted, because the commit hash is still being used, just now when manually creating the stabilisation branch.
- The rc1 release branch is reviewed and merged into the stabilisation branch.
- Any stabilisation PRs are merged into the `version-X.Y.0` branch, and more RCs if needed.
- Finally, a `release-vX.Y.0` script run is created, merged, and tagged.
- At this point, the `version-X.Y.0` branch is merged back into `master` via a PR (and can be deleted, as the tags retain the necessary state).

For hotfixes:
- Release manager creates a branch from the target tag named `hotfix-vX.Y.Z`, ensuring that `X.Y` matches the target tag, and pushes this to `zcash/zcash`.
- PRs are made onto that branch as usual.
- Release manager runs the release script on this branch, to create either `release-vX.Y.Z-rc1` or `release-vX.Y.Z` (depending on whether we plan to stabilise the hotfix, or if we have already done so privately for e.g. a security fix). This gets reviewed and merged as normal.
- Same as above for next release script run, except now it's either `release-vX.Y.Z-rc2` or `release-vX.Y.Z` (and so on).